app/publishing/publish-statement.js

```diff
@@ -1,14 +1,27 @@
 const { EMAIL } = require('../constants/methods')
 
+const { getExistingDocument } = require('../processing/publish')
+const validateEmail = require('./validate-email')
 const getPersonalisation = require('./get-personalisation')
-const handlePublishReasoning = require('./handle-publish-reasoning')
 const publish = require('./publish')
+const handlePublishReasoning = require('./handle-publish-reasoning')
 const saveRequest = require('./save-request')
-const validateEmail = require('./validate-email')
 
 const publishStatement = async (request) => {
   let reason
   let response
+
+  try {
+    const existingDocument = await getExistingDocument(request.documentReference)
+    if (existingDocument) {
+      console.info(`Duplicate document received, skipping ${existingDocument.documentReference}`)
+      return
+    }
+  } catch (err) {
+    console.log('Could not check for duplicates')
+    return
+  }
+
```

Comment:
You're right, returning from this will cause the message to be completed.

So reason for breaking it out into its own try-catch was to handle what happens when Sequelize breaks trying to read its table. I originally had it in the main try-catch but because that has logic in its catch and finally statements.

The pattern I was looking to adopt was, if it has a duplicate we don't process anymore or if it cannot check for a duplicate. But this needs to have replay ability.

I think throwing an error if there's a Sequelize problem will cause parent function to go into catch and so long as it doesn't have a category of validation, will not be cleared from the Topic.